feat(agent): implement comprehensive error handling for schema validation

FunamaYukina opened the pull request:
## Issue

- ref: https://github.com/route06/liam-internal/issues/5543

## Why is this change needed?

Previously, schema validation errors weren't displayed to users because:

1. **Missing message ID**: `AIMessage` instances lacked the `id` property required by `MessageTupleManager.add()`
2. **Message processing failure**: Without valid IDs, `messageManagerRef.current.add()` returned `null`, preventing real-time display
3. **No error persistence**: Users couldn't see validation errors when reloading the page
4. **Testing compatibility**: JSON formatting needed to match test expectations

## Changes

### Real-time Error Streaming
- **Add randomUUID() for message IDs**: Both schema deparser and database execution error messages now include unique IDs
- **Enable real-time streaming**: `dispatchCustomEvent` now works correctly with valid message IDs
- **Maintain LangSmith error recording**: `WorkflowTerminationError` still properly records workflow failures
- **Fix test compatibility**: Adjusted JSON formatting to match existing test expectations

### Error Display UI Components  
- **ErrorDisplay component**: Created reusable error display component following project patterns
- **Type-safe error handling**: Enhanced `useStream` hook with proper TypeScript error handling
- **Integrated error states**: Chat component now displays both streaming and persistent errors

### Checkpoint Error Retrieval
- **LangGraph native API usage**: Implemented `getCheckpointErrors` using `checkpointer.getTuple()` for accessing `__error__` channel data
- **PendingWrites access**: Discovered that `__error__` channel data is available through `pendingWrites` in checkpoint tuples
- **Simplified implementation**: Leverages `SupabaseCheckpointSaver`'s built-in decoding capabilities instead of custom hex/Base64 processing
- **Repository integration**: Added `getCheckpointErrorData` method to schema repository interface (used as fallback)
- **Server-side error fetching**: SessionDetailPage now retrieves and displays persistent validation errors

### Code Simplification
- **Removed complex decoding**: Eliminated 80+ lines of hex/Base64/JSON decoding functions
- **Framework-native approach**: Uses LangGraph's standard API instead of direct database manipulation
- **Improved maintainability**: Cleaner, more readable code that follows LangGraph patterns

## Screenshots

https://liam-app-git-validate-schema-node-error-handling-re-2-liambx.vercel.app/design_sessions/ffa6ba8c-9eca-403a-bf43-ab6aeef9e4f7?deepModeling=true

https://github.com/user-attachments/assets/ae4deb95-79c5-4fbd-8c44-fdc7b29db31f



## LangSmith Log

https://smith.langchain.com/o/eed4d2d8-0bd8-4ca4-a452-4da88ef63fd6/projects/p/99d63e81-58f9-475a-8985-409b591e26b0?timeModel=%7B%22duration%22%3A%2214d%22%7D&peek=5af3a110-7e85-4bac-a88a-20c67c71e160&peeked_trace=5af3a110-7e85-4bac-a88a-20c67c71e160

<img width="893" height="82" alt="ss 3892" src="https://github.com/user-attachments/assets/55c308a7-6441-4d23-b71a-b6d561b55cb7" />

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Chat now surfaces clear error messages when streaming fails or the session has workflow issues.
  * Session details initialize with any existing workflow errors displayed immediately.
  * Streaming errors are handled without blocking other message updates.

* **Style**
  * Introduced an accessible, polished error alert in the chat, improving visibility and readability of issues.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

MH4GF commented on frontend/internal-packages/agent/src/workflow/nodes/validateInitialSchemaNode.ts:
```diff
@@ -1,27 +1,38 @@
+  userMessage: string,
+  errorMessage: string,
+): Promise<never> {
+  const aiMessage = new AIMessage({
+    id: randomUUID(),
+    content: userMessage,
+  })
+
+  await dispatchCustomEvent(SSE_EVENTS.MESSAGES, aiMessage)
```
In this implementation, since the message isn't saved in langgraph's state, I think it will disappear after reloading.

FunamaYukina replied:
@MH4GF 
You're absolutely right. I'm trying to separate the error message display process from the error information retrieval process during reloading.

Should I create a pull request after completing all investigations?🤔

> Next Pull Request
Retrieve error information during reloading
Improve error message display content and design

MH4GF reviewed (changes requested):
Sorry, please check.

MH4GF commented on frontend/internal-packages/agent/src/repositories/supabase.ts:
Please delete it since I'm not using it.

FunamaYukina commented:
https://github.com/liam-hq/liam/pull/3498/commits/a412f459c3925cb41e4b3e3bf9b876560f832b13
 at SessionDetailPage.tsx & SessionDetailPageClient.tsx 
  Renamed initialCheckpointError to initialWorkflowError for better readability

https://github.com/liam-hq/liam/pull/3498/commits/43e48e44823d58af9dec992e286c4127824f327d
 at getCheckpointErrors.ts, InMemoryRepository.ts,  supabase.ts, types.ts
 Removed unused `getCheckpointErrorData` method and replaced complex error extraction with simple type guards

https://github.com/liam-hq/liam/pull/3498/commits/d5f4b7cd6c65ed9e6f7514c6be176cade51bfc2f

at  getCheckpointErrors.ts, SessionDetailPage.tsx, index.ts - Simplified getCheckpointErrors to return string[] instead of complex error objects

MH4GF reviewed (approved):
LGTM 👍🏻 Thanks!!
